x/authz/keeper/grpc_query.go

```diff
@@ -13,39 +13,56 @@ import (
 	"github.com/cosmos/cosmos-sdk/store/prefix"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/cosmos/cosmos-sdk/types/query"
-	"github.com/cosmos/cosmos-sdk/x/authz/types"
+	"github.com/cosmos/cosmos-sdk/x/authz"
 )
 
-var _ types.QueryServer = Keeper{}
+var _ authz.QueryServer = Keeper{}
 
-// Authorizations implements the Query/Authorizations gRPC method.
-func (k Keeper) Authorizations(c context.Context, req *types.QueryAuthorizationsRequest) (*types.QueryAuthorizationsResponse, error) {
+// Authorizations implements the Query/Grants gRPC method.
+func (k Keeper) Grants(c context.Context, req *authz.QueryGrantsRequest) (*authz.QueryGrantsResponse, error) {
 	if req == nil {
 		return nil, status.Errorf(codes.InvalidArgument, "empty request")
 	}
 
 	granter, err := sdk.AccAddressFromBech32(req.Granter)
-
 	if err != nil {
 		return nil, err
 	}
-	grantee, err := sdk.AccAddressFromBech32(req.Grantee)
 
+	grantee, err := sdk.AccAddressFromBech32(req.Grantee)
 	if err != nil {
 		return nil, err
 	}
 	ctx := sdk.UnwrapSDKContext(c)
 
 	store := ctx.KVStore(k.storeKey)
-	key := types.GetAuthorizationStoreKey(grantee, granter, "")
+	key := grantStoreKey(grantee, granter, "")
 	authStore := prefix.NewStore(store, key)
-	var authorizations []*types.AuthorizationGrant
+
+	if req.MsgTypeUrl != "" {
+		authorization, expiration := k.GetCleanAuthorization(ctx, grantee, granter, req.MsgTypeUrl)
```

Comment:
Could we add a comment that since this is in a query, `ctx` is ephemeral so `k.GetCleanAuthorization()` doesn't actually delete the authorization from state

Reply:
it will depend on who is calling the method. If it's called in a Tx context (other msg in a tx context calling this method), then it may change the state.